src/tokens.js

```diff
@@ -345,48 +406,71 @@ const tokens = {
     }
 
     // Create new data
-    const newTxData = {'inputs': [input], 'outputs': outputs, 'tokens': [token]};
+    const newTxData = {'inputs': inputs, 'outputs': outputs, 'tokens': [token]};
     return newTxData;
   },
 
   /**
    * Mint new tokens
    *
-   * @param {string} txID Hash of the transaction to be used to mint tokens
-   * @param {number} index Index of the output being spent
-   * @param {string} addressSpent Address of the output being spent
+   * @param {Object} mintInput tx containing mint authority {
+   *   {string} tx_id Hash of the tx
+   *   {number} index Index of the output being spent
+   *   {string} address The address associated with (tx_id, index)
+   * }
    * @param {string} token Token uid to be minted
    * @param {string} address Address to receive the amount of the generated token
    * @param {number} amount Amount of the token that will be minted
-   * @param {string} pin Pin to generate new addresses, if necessary
-   * @param {Object} {
+   * @param {Object} depositInputs Inputs used for the HTR deposit (optional) {
+   *   {Object} inputs List of the inputs [{
+   *     {string} tx_id Hash of the tx
+   *     {number} index Index of the output
+   *     {string} address The address associated with (tx_id, index)
+   *     {number} token Token info, which should always be 0 in this case
+   *   }]
+   *   {number} amount Sum of the inputs. It might be more than the required deposit, so a change output will be created
+   * }
+   * @param {Object} options {
    *   {number} minimumTimestamp Tx minimum timestamp (default = 0)
    *   {boolean} createAnotherMint If should create another mint output after spending this one
    *   {boolean} createMelt If should create a melt output (useful when creating a new token)
    * }
    *
+   * @throws {InsufficientTokensError} If not enough tokens for deposit
+   *
    * @return {Promise} Promise that resolves when token is minted or an error from the backend arrives
    *
    * @memberof Tokens
    * @inner
    */
-  mintTokens(txID, index, addressSpent, token, address, amount, pin, options) {
+  mintTokens(mintInput, token, address, amount, depositInputs, pin, options) {
     const fnOptions = Object.assign({
       createAnotherMint: true,
       createMelt: false,
       minimumTimestamp: 0,
     }, options);
     // Get mint data
-    let newTxData = this.createMintData(txID, index, addressSpent, token, address, amount, fnOptions);
-    return transaction.sendTransaction(newTxData, pin, fnOptions);
+    const promise = new Promise((resolve, reject) => {
+      let newTxData;
+      try {
+        newTxData = this.createMintData(mintInput, token, address, amount, depositInputs, fnOptions);
+      } catch (e) {
```

Comment:
What error you are expecting to catch here? I didn't see any `throw` in the `createMintData ` method. 

And you should catch the specific error

Reply:
`createMintData ()` uses `getMintDepositInfo()`, which throws.

Changed it to catch specific error.